### ISSUE TITLE
Unable to construct scenerio

### ISSUE THREAD

**Issue by BhrugaShah** [interopion/pillbox-issues#9]
Pillbox app:
I create subject but unable to save and construct the scenario

**Comment by vasofilipov** [comment]
Hi,

Can you please describe in details what is the issue you are facing so that we can give you the best possible help.

Thanks,
Vaso

**Comment by BhrugaShah** [comment]
Well,
I create the subject and save it.
When I go to construct scenario- I Can’t find the subj in the drop down menu

Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
Hi again,
1. Have you tried to find if the subject (I assume subject = patient) in the patient browser https://pillbox.medapptech.com/patients ? What is the name/mrn of the patient you've registered?
2. Also do you only press the down icon on the patient field? That widget is a dropdown/search input combo and you would need to write something to find a patient - parts of the patient's first or last name, mrn or date of birth?

**Comment by BhrugaShah** [comment]
Yes, done both of that multiple time.

If you like I can share my screen with you and walk you through

Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
Sure. Google hangouts or something else?

**Comment by BhrugaShah** [comment]
I am on secured serve which protects Pt’s info.

Would you be able to set up webinar?

Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
I don't think I could at the moment, but then again we might not need to. I think the problem is because of MRN duplication. Can you please try to create a test patient, filling all the required data - first/last name, date of birth and MRN, something like;
First Name: Test
Last Name: Patient
MRN: TestMRNSep2017
Date of Birth: 1900-01-01

This would be to rule out the possibility of something wrong with your account. I strongly suspect that there is an issue with the data you're using to create the patient.

Thank you,
Vaso

**Comment by BhrugaShah** [comment]
Oh My God !
It works!

Can I put MRN as Subject ID?
I tried, MRN: TestMRN102001. It worked!

However, It doesn’t let me select my TestMRN1021001☹ to create the scenario.


Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
Yes, you can put Subject ID or something else as the MRN, it just has to be unique for the patient.
You can recognize when a patient is successfully created a delete button will appear on the page next to the save button - if not - there is a problem - probably the MRN isn't unique.

Thank you for reporting this. I hope we will be able to arrange a patch where such errors will be better communicated to the user - like when using a MRN for an already existing patient to alert that.

Regards,
Vaso

**Comment by BhrugaShah** [comment]
Well, I have tried MRN millions of time, it doesn’t work.
Still I cannot select the subject from the drop down menu  to enter the meds.

Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
I am able to select that patient with MRN "TestMRNSep2017", just by typing "testmrnsep" in the search input (the one that has a hint "Type to search..."). and then clicking on the patient from the dropdown.

**Comment by BhrugaShah** [comment]
Good Morning,

I am still waiting for your response.
My pt is coming on Monday and I would like the pillbox to work. ☹

Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
Good Morning @BhrugaShah,

May be my last message from yesterday got lost in all the updates in this thread. I've reviewed the logs for the application and it seems the problem with the patients you try to create is not filling the required info. I see that you've managed to create the test patient we've talked about yesterday and that patient seems like to be usable. What you can try is:

1. Log in pillbox using your account - https://pillbox.medapptech.com
2. Verify that you can see the patient by visiting the patient browser page - https://pillbox.medapptech.com/patients 
3. Go back to the main page - https://pillbox.medapptech.com
4. From the main page, in the panel that is called "Patient" there is a search input field.
5. In this field type in the patient's MRN (testmrnsep2017)
6. The patient should appear in the dropdown
7. Click on the patient
8. In the Medication panel below a single medication should populate the list
9. Click on add/remove medication buttons to chose the medications you'd like to have chosen
10. Use the big green button to launch an exercise.

If all of this works for that patient you can create a new patient - go to the create new patient page - https://pillbox.medapptech.com/patients/new 
It is important to fill all the fields, if cannot provide real names use something else in place of the name like HIDDEN, Jan 1st 1900 for date of birth. As for the MRN - anything unique, preferably without spaces and special characters should be fine.

By the way what browser and version are you using?

Regards,
Vaso

**Comment by BhrugaShah** [comment]
Good Afternoon,

A little bit progress with the help of  following directions.:)
Able to add meds Good! Yay!!! But cannot find many meds that the subject is taking. ☹
How can I add meds those are not on the list?
Also, Quantity is only goes up to 2 tab. Pt with diabetes takes insulin up to 55 or 100 units. We should be able to add more units.
Able to get the PIN number, Yay!, however unable to launch another browser where I can enter the  Pin code to start the exercise.


Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
Hello @BhrugaShah,

I am very glad to hear you are making progress.
Adding new medications is possible - you can go to the add medication page - https://pillbox.medapptech.com/meds/new where you will have a form for adding a new medication. Only name and rxnorm code are mandatory fields and also those fields are linked with an API with the RXNorm codes database so adding shouldn't be too hard.

As for the launching of the actual pillbox exercise - you would need to either scan the QR code that appears below the window with a mobile device (preferably a tablet) or copy the link (right clicking on the QRCode and selecting copy link or copy link location from the context menu).

Changing the allowed quantities is something that hasn't a practical short term solution, but it would definitely be something worth consideration and implementation in the next release of pillbox. The unpractical solution would be to create a medication that carries the information about the quantity in the name like - "Medication X - 120 units" and having this marked as one dosage for 120 units, 1.5 for 180 units, 2 for 240.

Regards,
Vaso

**Comment by BhrugaShah** [comment]
Hi Vaso,

How can I add more meds in the subject already created and have had med assigned?
Thanks in advance.
I have pt coming in an hour, I add 5 meds, but need to add more meds before she gets here.


Thanks,
Bhruga Shah MS, BSN-RN
Clinical Research Nurse,
Inova Heart and Vascular Institute
T 703.776.2995|
Email Bhruga.shah@inova.org<mailto:Bhruga.shah@inova.org>

**Comment by vasofilipov** [comment]
Hello @BhrugaShah,
You can add or remove meds for new exercises the same way you first assigned any medications:

1. Login to pillbox with your account;
2. Use the search input/dropdown to select the relevant patient (by typing that patient's MRN);
3. In the medications panel - click on the yellow button (Add/Remove) - this will open the add/remove popup window;
4. Navigate the medications using the next/previous buttons or start typing the medication name or RXNorm code in the search input on the top of the popup;
5. Check or uncheck any medications you wish;
6. Click the use selected button to close the popup;
7 Continue the work flow by assigning the newly added medications intervals and quantity.

I hope this will help.

Regards,
Vaso
